libs/blocks/table/table.js

```diff
@@ -94,7 +93,18 @@ function handleHeading(table, headingCols) {
       col.setAttribute('role', 'columnheader');
     }
 
-    nodeToApplyRoleScope.setAttribute('scope', 'col');
+    const headings = col.querySelectorAll('h1, h2, h3, h4, h5, h6');
+
+    headings.forEach((heading) => {
```

Comment:
nit: no need to cache and saves 1 line
```suggestion
    col.querySelectorAll('h1, h2, h3, h4, h5, h6').forEach((heading) => {
```